### PR TITLE
Add `variant` prop to Heading

### PR DESCRIPTION
**Pull request by langermank**
UI mob pair session 💅 

#### New

- Adds new `variant` prop for styling small, medium, and large headings

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: e2e/components/Heading.test.ts [hunk 1]
```diff
@@ -1,33 +1,74 @@
...
-        })
-      })
-    }
+    test('default @vrt', async ({page}) => {
+      await visit(page, {
+        id: 'components-heading--default',
+      })
+
+      // Default state
+      expect(await page.screenshot()).toMatchSnapshot(`Heading.Default.png`)
```
Comment (siddharthkp) [inline e2e/components/Heading.test.ts:12]: aww? we don't tests for all themes?

File: packages/react/src/Heading/Heading.docs.json [hunk 2]
```diff
@@ -14,6 +14,11 @@
       "name": "as",
       "type": "React.ElementType",
       "defaultValue": "\"h2\""
+    },
+    {
+      "name": "variant",
+      "type": "'large' | 'medium' | 'small'",
+      "defaultValue": ""
```
Comment (siddharthkp) [inline packages/react/src/Heading/Heading.docs.json:21]:
```suggestion

```

there is no default value. (this suggests default value is empty string)

File: packages/react/src/Heading/Heading.featires.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
```
Comment (siddharthkp) [inline packages/react/src/Heading/Heading.featires.stories.tsx:1]: typo in file name, would mess with automated examples in docs

```diff
- Heading.featires.stories.tsx
+ Heading.features.stories.tsx
```

File: .playwright/snapshots/components/Blankslate.test.ts-snapshots/Blankslate-Default-dark-colorblind-linux.png
Comment (siddharthkp) [inline .playwright/snapshots/components/Blankslate.test.ts-snapshots/Blankslate-Default-dark-colorblind-linux.png:1]: Strange that these tests show up on this PR? Are these changes from a different PR committed here?

**Review by siddharthkp — CHANGES_REQUESTED** [review]
Looks good, suggested some cleanup

File: e2e/components/Heading.test.ts (hunk 1 above)
Comment (langermank) [reply e2e/components/Heading.test.ts:12]: Its just testing for font-size so color doesn't matter 😄

File: packages/react/src/Heading/Heading.featires.stories.tsx (hunk 3 above)
Comment (langermank) [reply packages/react/src/Heading/Heading.featires.stories.tsx:1]: Tire 😄 thanks!

File: .changeset/clever-birds-press.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@primer/react": patch
```
Comment (joshblack) [inline .changeset/clever-birds-press.md:2]:
```suggestion
"@primer/react": minor
```

**Review by joshblack — APPROVED** [review]

**Review by mperrotti — APPROVED** [review]

**Pull request merged**